app/views/RoomView/UploadProgress.tsx

```diff
@@ -51,23 +52,26 @@ const styles = StyleSheet.create({
 	}
 });
 
-class UploadProgress extends Component {
-	static propTypes = {
-		width: PropTypes.number,
-		rid: PropTypes.string,
-		theme: PropTypes.string,
-		user: PropTypes.shape({
-			id: PropTypes.string.isRequired,
-			username: PropTypes.string.isRequired,
-			token: PropTypes.string.isRequired
-		}),
-		baseUrl: PropTypes.string.isRequired
-	};
+interface IUploadProgressProps {
+	width: number;
+	rid: string;
+	user: Partial<Pick<IUser, 'id' | 'username' | 'token'>>;
```

Comment:
Wouldn't it be better to just use Pick?
Or these properties are not of the same type as those of the IUser, mandatory or something like that.